modules/pam_echo/pam_echo.c

```diff
@@ -182,6 +183,12 @@ pam_echo (pam_handle_t *pamh, int flags, int argc, const char **argv)
 	  return PAM_IGNORE;
 	}
 
+      if ((uint64_t) st.st_size >= (uint64_t) SIZE_MAX)
```

Comment:
Let's not assume that uint64_t is big enough and use uintmax_t instead, or cast SIZE_MAX to __typeof__(st.st_size).

Reply:
SIZE_MAX is very large anyway. During our reviews we've encountered a sane limit in pam_motd, so I have copied it now.

Reply:
SIZE_MAX is way too large indeed, but returning PAM_BUF_ERR on sizes greater than 0x10000 is a change of behavior.

Reply:
Adjusted as recommended.